<a href="https://colab.research.google.com/github/chirag3096/Project/blob/main/GO_DIGITAL_ASSIGNMENTS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [76]:
# importing pandas
import pandas as pd

In [77]:
# Reading the CSV file 
employee_data = pd.read_csv("/content/employee_data.csv")
insurance_data = pd.read_csv("/content/insurance_data.csv")
vendor_data = pd.read_csv("/content/vendor_data.csv")

In [78]:
employee_data.head()

AGENT_ID      AGENT_NAME DATE_OF_JOINING          ADDRESS_LINE1  \
0  AGENT00001       Ray Johns      1993-06-05       1402 Maggies Way   
1  AGENT00002   Angelo Borjon      2005-12-27         414 Tanya Pass   
2  AGENT00003  Candy Spellman      2003-09-02    606 National Street   
3  AGENT00004      Mary Smith      2004-09-23  235 Hugh Thomas Drive   
4  AGENT00005    Mildred Diaz      2011-06-21  3426 Broadview Street   

  ADDRESS_LINE2              CITY STATE  POSTAL_CODE  EMP_ROUTING_NUMBER  \
0           NaN  Waterbury Center    VT         5677            34584958   
1           NaN       Panama City    FL        32404           107363763   
2          #306      Fayetteville    AR        72701            81744097   
3           NaN       Panama City    FL        32404            67563771   
4           NaN        Montgomery    AL        36110           114951317   

      EMP_ACCT_NUMBER  
0  HKUN51252328472585  
1  OPIS19290040088204  
2  YSCJ67489688482590  
3  ZANG21285355574581  
4  DZFS82244494451134

In [81]:
insurance_data.head()

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
0  2020-06-01 00:00:00    TXN00000001   A00003822   PLC00008468    2015-06-23   
1  2020-06-01 00:00:00    TXN00000002   A00008149   PLC00009594    2018-04-21   
2  2020-06-01 00:00:00    TXN00000003   A00003172   PLC00007969    2019-10-03   
3  2020-06-01 00:00:00    TXN00000004   A00007572   PLC00009292    2016-11-29   
4  2020-06-01 00:00:00    TXN00000005   A00008173   PLC00000204    2011-12-26   

      LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  ...  \
0  2020-05-16  2020-05-21         Health          157.13          9000  ...   
1  2020-05-13  2020-05-18       Property          141.71         26000  ...   
2  2020-05-21  2020-05-26       Property          157.24         13000  ...   
3  2020-05-14  2020-05-19         Health          172.87         16000  ...   
4  2020-05-17  2020-05-22         Travel           88.53          3000  ...   

  CLAIM_STATUS INCIDENT_SEVERITY AUTHORITY_CONTACTED ANY_INJURY  \
0            A        Major Loss              Police          0   
1            A        Total Loss           Ambulance          1   
2            A        Total Loss              Police          0   
3            A        Minor Loss           Ambulance          0   
4            A        Major Loss              Police          0   

  POLICE_REPORT_AVAILABLE  INCIDENT_STATE   INCIDENT_CITY  \
0                       1              GA        Savannah   
1                       0              AL      Montgomery   
2                       1              CO  Grand Junction   
3                       0              GA        Savannah   
4                       1              TN       Nashville   

  INCIDENT_HOUR_OF_THE_DAY    AGENT_ID  VENDOR_ID  
0                        4  AGENT00413  VNDR00556  
1                        0  AGENT00769  VNDR00592  
2                       19  AGENT00883  VNDR00031  
3                       12  AGENT00278  VNDR00075  
4                       18  AGENT00636  VNDR00472  

[5 rows x 38 columns]

In [6]:
vendor_data.head()

VENDOR_ID              VENDOR_NAME                   ADDRESS_LINE1  \
0  VNDR00001  King, Proctor and Jones        2027 North Shannon Drive   
1  VNDR00002               Garcia Ltd          5701 East Shirley Lane   
2  VNDR00003               Cherry LLC            1217 Cottondale Road   
3  VNDR00004              Mays-Benson  227 West Montgomery Cross Road   
4  VNDR00005               Wilson PLC            23 North Hill Street   

  ADDRESS_LINE2          CITY STATE  POSTAL_CODE  
0            #5  Fayetteville    AR        72703  
1           NaN    Montgomery    AL        36117  
2           NaN    Montgomery    AL        36109  
3          #736      Savannah    GA        31406  
4           NaN     Nashville    TN        37210

**Task # 1**

Merge the 3 dataset and create 1 view of data.
You can merge insurance_data.csv and employee_data.csv on
AGENT_ID You can merge insurance_data.csv and vendor_data.csv on
VENDOR_ID Note: Use left Outer join as not all claims require Vendor


In [27]:
# Mergeing the employee_data & insurance_data CSV files
data_1=pd.merge(insurance_data,employee_data,on="AGENT_ID",how="left")

In [8]:
data_1.head()

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER POLICY_EFF_DT  \
0  2020-06-01 00:00:00    TXN00000001   A00003822   PLC00008468    2015-06-23   
1  2020-06-01 00:00:00    TXN00000002   A00008149   PLC00009594    2018-04-21   
2  2020-06-01 00:00:00    TXN00000003   A00003172   PLC00007969    2019-10-03   
3  2020-06-01 00:00:00    TXN00000004   A00007572   PLC00009292    2016-11-29   
4  2020-06-01 00:00:00    TXN00000005   A00008173   PLC00000204    2011-12-26   

      LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  CLAIM_AMOUNT  ...  \
0  2020-05-16  2020-05-21         Health          157.13          9000  ...   
1  2020-05-13  2020-05-18       Property          141.71         26000  ...   
2  2020-05-21  2020-05-26       Property          157.24         13000  ...   
3  2020-05-14  2020-05-19         Health          172.87         16000  ...   
4  2020-05-17  2020-05-22         Travel           88.53          3000  ...   

   VENDOR_ID       AGENT_NAME DATE_OF_JOINING              ADDRESS_LINE1_y  \
0  VNDR00556      Amy Wangler      1994-01-06          584 Rural Hill Road   
1  VNDR00592   Jessica Montez      2004-05-28        6114 West Glenn Drive   
2  VNDR00031     Howard Smith      2004-08-01  2500 Snowdoun Chambers Road   
3  VNDR00075  Patricia Tucker      2007-06-12           313 Lone Oak Drive   
4  VNDR00472      Brenda Bean      2010-01-19             120 Wells Avenue   

  ADDRESS_LINE2_y       CITY_y STATE_y POSTAL_CODE_y  EMP_ROUTING_NUMBER  \
0             NaN    Nashville      TN         37217            35441155   
1              #1     Glendale      AZ         85301            44055451   
2             NaN   Montgomery      AL         36105            55770246   
3             NaN       Norman      OK         73071            87652976   
4             NaN  Glen Burnie      MD         21061            76119019   

      EMP_ACCT_NUMBER  
0  LRZO53254369804934  
1  XBYG14382904542874  
2  WRXH14597428306333  
3  YEPN16077902472499  
4  HZBK86544578578547  

[5 rows x 47 columns]

In [28]:
## Mergeing the data_1 & Vendor_data CSV files
view=pd.merge(data_1,vendor_data,on="VENDOR_ID",how="left")
view

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER  \
0     2020-06-01 00:00:00    TXN00000001   A00003822   PLC00008468   
1     2020-06-01 00:00:00    TXN00000002   A00008149   PLC00009594   
2     2020-06-01 00:00:00    TXN00000003   A00003172   PLC00007969   
3     2020-06-01 00:00:00    TXN00000004   A00007572   PLC00009292   
4     2020-06-01 00:00:00    TXN00000005   A00008173   PLC00000204   
...                   ...            ...         ...           ...   
9995  2021-06-30 00:00:00    TXN00009996   A00006029   PLC00007245   
9996  2021-06-30 00:00:00    TXN00009997   A00003555   PLC00002590   
9997  2021-06-30 00:00:00    TXN00009998   A00004230   PLC00006613   
9998  2021-06-30 00:00:00    TXN00009999   A00006694   PLC00001781   
9999  2021-06-30 00:00:00    TXN00010000   A00000390   PLC00005197   

     POLICY_EFF_DT     LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  \
0       2015-06-23  2020-05-16  2020-05-21         Health          157.13   
1       2018-04-21  2020-05-13  2020-05-18       Property          141.71   
2       2019-10-03  2020-05-21  2020-05-26       Property          157.24   
3       2016-11-29  2020-05-14  2020-05-19         Health          172.87   
4       2011-12-26  2020-05-17  2020-05-22         Travel           88.53   
...            ...         ...         ...            ...             ...   
9995    2014-04-11  2021-06-25  2021-06-26          Motor          105.75   
9996    2015-08-25  2021-06-12  2021-06-17           Life           64.59   
9997    2018-10-11  2021-06-18  2021-06-23         Health          183.50   
9998    2016-01-29  2021-06-16  2021-06-21          Motor          113.46   
9999    2014-09-17  2021-06-21  2021-06-24         Mobile            8.22   

      CLAIM_AMOUNT  ... STATE_y POSTAL_CODE_y EMP_ROUTING_NUMBER  \
0             9000  ...      TN         37217           35441155   
1            26000  ...      AZ         85301           44055451   
2            13000  ...      AL         36105           55770246   
3            16000  ...      OK         73071           87652976   
4             3000  ...      MD         21061           76119019   
...            ...  ...     ...           ...                ...   
9995          1000  ...      AR         72703           35413882   
9996         67000  ...      GA         31405           91206790   
9997          8000  ...      KY         40272          115241097   
9998          2000  ...      AR         72703           13995380   
9999           300  ...      OK         73003           39790996   

         EMP_ACCT_NUMBER                       VENDOR_NAME  \
0     LRZO53254369804934                        Ramsey Inc   
1     XBYG14382904542874  Morales, Campbell and Washington   
2     WRXH14597428306333                       Johnson Ltd   
3     YEPN16077902472499                  Blackwell-Foster   
4     HZBK86544578578547                       Evans-Scott   
...                  ...                               ...   
9995  LFOJ76068590051807                               NaN   
9996  AHEF60771774635030          Newton, Hodges and Henry   
9997  WREH18805592939657                    Reese-Mckinney   
9998  AAZI21883736448048       Harmon, Manning and Bradley   
9999  RPES05613584687957                    Foster-Sanchez   

                  ADDRESS_LINE1 ADDRESS_LINE2          CITY  STATE  \
0          3114 West 20th Court           NaN   Panama City     FL   
1         42 West Louise Street             B  Fayetteville     AR   
2       2619 North Quality Lane          #315  Fayetteville     AR   
3             3609 Meadow Drive           NaN     Nashville     TN   
4     4001 19th Place Northeast           NaN    Washington     DC   
...                         ...           ...           ...    ...   
9995                        NaN           NaN           NaN    NaN   
9996      32512 Christine Drive           NaN    Union City     CA   
9997         20 Garfield Street           NaN       Ne

**Task #2**

Business Leader wants to find Top 3 Insurance Type where we are getting
most insurance claims?

In [10]:
x=insurance_data.INSURANCE_TYPE.value_counts()
x[:3]

Property    1692
Mobile      1692
Health      1690
Name: INSURANCE_TYPE, dtype: int64

Top 3 Insurance Type where we are getting most insurance claims are :
       
       Property  
       Mobile    
       Health 


**Task #3**

Business Leader wants to find Top 5 States where we are getting most insurance claims
for customer belonging to HIGH(H) risk segment?

In [11]:
y=insurance_data[insurance_data['RISK_SEGMENTATION']=='H'].STATE.value_counts()
y[:5]

CA    148
AZ    105
FL    104
TN    100
AR     97
Name: STATE, dtype: int64

Top 5 States where we are getting most insurance claims for customer belonging to HIGH(H) risk segment are :

     CA   
     AZ    
     FL    
     TN    
     AR     

**Task #4**

Business wants to create a new variable “COLOCATION” which will have following values IF
Customer State == Incident State == Agent Address State THEN 1 ELSE 0
Find the mean of this new column

In [12]:
view.columns

Index(['TXN_DATE_TIME', 'TRANSACTION_ID', 'CUSTOMER_ID', 'POLICY_NUMBER',
       'POLICY_EFF_DT', 'LOSS_DT', 'REPORT_DT', 'INSURANCE_TYPE',
       'PREMIUM_AMOUNT', 'CLAIM_AMOUNT', 'CUSTOMER_NAME', 'ADDRESS_LINE1_x',
       'ADDRESS_LINE2_x', 'CITY_x', 'STATE_x', 'POSTAL_CODE_x', 'SSN',
       'MARITAL_STATUS', 'AGE', 'TENURE', 'EMPLOYMENT_STATUS',
       'NO_OF_FAMILY_MEMBERS', 'RISK_SEGMENTATION', 'HOUSE_TYPE',
       'SOCIAL_CLASS', 'ROUTING_NUMBER', 'ACCT_NUMBER',
       'CUSTOMER_EDUCATION_LEVEL', 'CLAIM_STATUS', 'INCIDENT_SEVERITY',
       'AUTHORITY_CONTACTED', 'ANY_INJURY', 'POLICE_REPORT_AVAILABLE',
       'INCIDENT_STATE', 'INCIDENT_CITY', 'INCIDENT_HOUR_OF_THE_DAY',
       'AGENT_ID', 'VENDOR_ID', 'AGENT_NAME', 'DATE_OF_JOINING',
       'ADDRESS_LINE1_y', 'ADDRESS_LINE2_y', 'CITY_y', 'STATE_y',
       'POSTAL_CODE_y', 'EMP_ROUTING_NUMBER', 'EMP_ACCT_NUMBER', 'VENDOR_NAME',
       'ADDRESS_LINE1', 'ADDRESS_LINE2', 'CITY', 'STATE', 'POSTAL_CODE'],
      dtype='object')

In [30]:
view['COLOCATION'] = (view['STATE_x']==view['INCIDENT_STATE'])&(view['INCIDENT_STATE']==view['STATE_y'])

In [25]:
view['COLOCATION']=view.COLOCATION.astype('int')

In [26]:
view.COLOCATION.mean()

0.0044

The mean of this new column COLOCATION = 0.0044

**Task #5**

Data entry error was detected in the data and you are required to correct it. If for any
claim transaction “AUTHORITY_CONTACTED” is NOT “Police” and POLICE_AVAILABLE == 1
Then Update “AUTHORITY_CONTACTED” to Police.

In [32]:
non_police = view[view['AUTHORITY_CONTACTED']!='Police']

In [33]:
non_police 

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER  \
1     2020-06-01 00:00:00    TXN00000002   A00008149   PLC00009594   
3     2020-06-01 00:00:00    TXN00000004   A00007572   PLC00009292   
5     2020-06-01 00:00:00    TXN00000006   A00003571   PLC00007244   
9     2020-06-01 00:00:00    TXN00000010   A00002020   PLC00007657   
10    2020-06-01 00:00:00    TXN00000011   A00007952   PLC00003007   
...                   ...            ...         ...           ...   
9994  2021-06-30 00:00:00    TXN00009995   A00003508   PLC00009242   
9995  2021-06-30 00:00:00    TXN00009996   A00006029   PLC00007245   
9997  2021-06-30 00:00:00    TXN00009998   A00004230   PLC00006613   
9998  2021-06-30 00:00:00    TXN00009999   A00006694   PLC00001781   
9999  2021-06-30 00:00:00    TXN00010000   A00000390   PLC00005197   

     POLICY_EFF_DT     LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  \
1       2018-04-21  2020-05-13  2020-05-18       Property          141.71   
3       2016-11-29  2020-05-14  2020-05-19         Health          172.87   
5       2012-12-28  2020-05-20  2020-05-25           Life           87.02   
9       2015-12-19  2020-05-15  2020-05-20         Travel           71.29   
10      2019-03-30  2020-05-20  2020-05-25           Life           61.45   
...            ...         ...         ...            ...             ...   
9994    2011-10-01  2021-06-24  2021-06-26       Property          135.33   
9995    2014-04-11  2021-06-25  2021-06-26          Motor          105.75   
9997    2018-10-11  2021-06-18  2021-06-23         Health          183.50   
9998    2016-01-29  2021-06-16  2021-06-21          Motor          113.46   
9999    2014-09-17  2021-06-21  2021-06-24         Mobile            8.22   

      CLAIM_AMOUNT  ... POSTAL_CODE_y EMP_ROUTING_NUMBER     EMP_ACCT_NUMBER  \
1            26000  ...         85301           44055451  XBYG14382904542874   
3            16000  ...         73071           87652976  YEPN16077902472499   
5            63000  ...         93311           35911575  VHQA45663806811975   
9             4000  ...         99508           28122449  GETW44783556180552   
10           92000  ...         36108          123461076  YIAT74328087165676   
...            ...  ...           ...                ...                 ...   
9994         34000  ...         40245           88548490  VZTY16125395831114   
9995          1000  ...         72703           35413882  LFOJ76068590051807   
9997          8000  ...         40272          115241097  WREH18805592939657   
9998          2000  ...         72703           13995380  AAZI21883736448048   
9999           300  ...         73003           39790996  RPES05613584687957   

                           VENDOR_NAME          ADDRESS_LINE1  ADDRESS_LINE2  \
1     Morales, Campbell and Washington  42 West Louise Street              B   
3                     Blackwell-Foster      3609 Meadow Drive            NaN   
5                                  NaN                    NaN            NaN   
9                                  NaN                    NaN            NaN   
10                      Foster-Sanchez    4255 Tanager Common            NaN   
...                                ...                    ...            ...   
9994                           Lee PLC    1650 Garland Avenue            NaN   
9995                               NaN                    NaN            NaN   
9997                    Reese-Mckinney     20 Garfield Street            NaN   
9998       Harmon, Manning and Bradley       1823 Dinger Road            NaN   
9999                    Foster-Sanchez    4255 Tanager Common            NaN   

              CITY STATE  POSTAL_CODE  COLOCATION  
1     Fayetteville    AR      72701.0       False  
3        Nashville    TN      37215.0       False  
5              NaN   NaN          NaN       False  
9              NaN   NaN          NaN       False  
10         Fremont    CA      94555.0       False  
...         

In [34]:
update= non_police[non_police['POLICE_REPORT_AVAILABLE']==1].index

In [35]:
update

Int64Index([   5,   10,   11,   12,   16,   18,   19,   21,   22,   28,
            ...
            9975, 9977, 9979, 9984, 9990, 9991, 9993, 9995, 9997, 9998],
           dtype='int64', length=4870)

In [36]:
view.loc[update,'AUTHORITY_CONTACTED'] = 'Police'

In [37]:
view[(view['POLICE_REPORT_AVAILABLE']==1)&(view['AUTHORITY_CONTACTED']!='Police')]

Empty DataFrame
Columns: [TXN_DATE_TIME, TRANSACTION_ID, CUSTOMER_ID, POLICY_NUMBER, POLICY_EFF_DT, LOSS_DT, REPORT_DT, INSURANCE_TYPE, PREMIUM_AMOUNT, CLAIM_AMOUNT, CUSTOMER_NAME, ADDRESS_LINE1_x, ADDRESS_LINE2_x, CITY_x, STATE_x, POSTAL_CODE_x, SSN, MARITAL_STATUS, AGE, TENURE, EMPLOYMENT_STATUS, NO_OF_FAMILY_MEMBERS, RISK_SEGMENTATION, HOUSE_TYPE, SOCIAL_CLASS, ROUTING_NUMBER, ACCT_NUMBER, CUSTOMER_EDUCATION_LEVEL, CLAIM_STATUS, INCIDENT_SEVERITY, AUTHORITY_CONTACTED, ANY_INJURY, POLICE_REPORT_AVAILABLE, INCIDENT_STATE, INCIDENT_CITY, INCIDENT_HOUR_OF_THE_DAY, AGENT_ID, VENDOR_ID, AGENT_NAME, DATE_OF_JOINING, ADDRESS_LINE1_y, ADDRESS_LINE2_y, CITY_y, STATE_y, POSTAL_CODE_y, EMP_ROUTING_NUMBER, EMP_ACCT_NUMBER, VENDOR_NAME, ADDRESS_LINE1, ADDRESS_LINE2, CITY, STATE, POSTAL_CODE, COLOCATION]
Index: []

[0 rows x 54 columns]

Task #6

Business wants to check the Claim Amount for deviation for each transaction, they would
like you to calculate as follow
CLAIM_DEVIATION = AVG_CLAIM_AMOUNT_FOR_LAST_30DAYS (same insurance type)
/ CURRENT_CLAIM_AMOUNT
If the value < 0.5 THEN CLAIM_DEVIATION = 1 ELSE 0
If there is less than 30 days of transaction history THEN
-1 Note: LAST_30DAYS does not include current day

**Task #7**

Find All Agents who have worked on more than 2 types of Insurance Claims. Sort them by
Total Claim Amount Approved under them in descending order

In [38]:
count = view.groupby('AGENT_ID')['INSURANCE_TYPE'].nunique()
count

AGENT_ID
AGENT00001    4
AGENT00002    2
AGENT00003    6
AGENT00004    6
AGENT00005    4
             ..
AGENT01196    4
AGENT01197    4
AGENT01198    4
AGENT01199    5
AGENT01200    6
Name: INSURANCE_TYPE, Length: 1200, dtype: int64

In [39]:
agent_insurance_types=count>2
agent_insurance_types

AGENT_ID
AGENT00001     True
AGENT00002    False
AGENT00003     True
AGENT00004     True
AGENT00005     True
              ...  
AGENT01196     True
AGENT01197     True
AGENT01198     True
AGENT01199     True
AGENT01200     True
Name: INSURANCE_TYPE, Length: 1200, dtype: bool

In [40]:
more_type_agents = agent_insurance_types[agent_insurance_types].index
more_type_agents

Index(['AGENT00001', 'AGENT00003', 'AGENT00004', 'AGENT00005', 'AGENT00006',
       'AGENT00007', 'AGENT00008', 'AGENT00009', 'AGENT00010', 'AGENT00011',
       ...
       'AGENT01191', 'AGENT01192', 'AGENT01193', 'AGENT01194', 'AGENT01195',
       'AGENT01196', 'AGENT01197', 'AGENT01198', 'AGENT01199', 'AGENT01200'],
      dtype='object', name='AGENT_ID', length=1164)

In [44]:
Top_agents = view.groupby(['AGENT_ID','AGENT_NAME']).sum().loc[more_type_agents][['CLAIM_AMOUNT']].sort_values('CLAIM_AMOUNT',ascending=False)

In [45]:
Top_agents

CLAIM_AMOUNT
AGENT_ID   AGENT_NAME                    
AGENT00807 Don Filkins             528800
AGENT00679 Clara Barnett           489000
AGENT00771 Roger Burns             422100
AGENT00125 Salvador Soriano        400400
AGENT00789 Alison Baron            392900
...                                   ...
AGENT00910 Vincent Ragan            15500
AGENT00689 Connie Hunt              13700
AGENT00732 Sylvia Tran              11300
AGENT00706 Laura Staggs             10800
AGENT00604 Thelma Salinas            5500

[1164 rows x 1 columns]

In [46]:
# Agents who have worked on more than 2 types of Insurance Claims
Top_agents.join(count)

CLAIM_AMOUNT  INSURANCE_TYPE
AGENT_ID   AGENT_NAME                                    
AGENT00807 Don Filkins             528800               6
AGENT00679 Clara Barnett           489000               4
AGENT00771 Roger Burns             422100               5
AGENT00125 Salvador Soriano        400400               4
AGENT00789 Alison Baron            392900               4
...                                   ...             ...
AGENT00910 Vincent Ragan            15500               3
AGENT00689 Connie Hunt              13700               3
AGENT00732 Sylvia Tran              11300               3
AGENT00706 Laura Staggs             10800               3
AGENT00604 Thelma Salinas            5500               3

[1164 rows x 2 columns]

**Task #8**

Mobile & Travel Insurance premium are discounted by 10%
Health and Property Insurance premium are increased by 7%
Life and Motor Insurance premium are marginally increased by 2%
What will be overall change in % of the Premium Amount Collected for all these Customer?

In [ ]:
premium_collected = view['PREMIUM_AMOUNT'].sum()

In [ ]:
prem_by_types = view.groupby('INSURANCE_TYPE').sum()['PREMIUM_AMOUNT']

In [ ]:
prem_by_types

INSURANCE_TYPE
Health       17671.9011
Life          2512.4244
Mobile       12305.0502
Motor         3307.8396
Property     14160.0081
Travel      100553.2137
Name: PREMIUM_AMOUNT, dtype: float64

**Task #9**

Business wants to give discount to customer who are loyal and under stress due to Covid 19. They have laid down an eligibility Criteria as follow
IF CUSTOMER_TENURE > 60 AND EMPLOYMENT_STATUS = “N”
AND NO_OF_FAMILY_MEMBERS >=4 THEN 1 ELSE 0 Create a new column “ELIGIBLE_FOR_DISCOUNT” and find it
mean

In [48]:
disc_status=(view['TENURE']>60)&(view['EMPLOYMENT_STATUS']=='N')&(view['NO_OF_FAMILY_MEMBERS']>=4)

In [49]:
view['ELIGIBLE_FOR_DISCOUNT'] = disc_status.astype('int')

In [51]:
# A new column “ELIGIBLE_FOR_DISCOUNT” 
view.ELIGIBLE_FOR_DISCOUNT

0       0
1       0
2       0
3       0
4       0
       ..
9995    0
9996    0
9997    0
9998    0
9999    0
Name: ELIGIBLE_FOR_DISCOUNT, Length: 10000, dtype: int64

In [52]:
view.ELIGIBLE_FOR_DISCOUNT.mean()

0.0299

A new column “ELIGIBLE_FOR_DISCOUNT” mean  = 0.0299

**Task #10**

Business wants to check Claim Velocity which is defined as follow
CLAIM_VELOCITY = NO_OF_CLAIMS_IN_LAST30DAYS (for the current insurance type)
/ NO_OF_CLAIMS_IN_LAST3DAYS (for the current insurance type)
Note: LAST30DAYS & LAST3DAYS does not include current
day

**Task #11**

Find all low performing agents i.e. employees who are in the bottom 5 percentile based
on Claims worked by them.


In [54]:
bottom_5 = view.groupby('AGENT_ID').sum().quantile(0.05)['CLAIM_AMOUNT']

In [55]:
Agent_grouped = view.groupby('AGENT_ID').sum()

In [56]:
low_performing_agents = Agent_grouped[Agent_grouped['CLAIM_AMOUNT']<=bottom_5][['CLAIM_AMOUNT']]

In [57]:
# low_performing_agents
low_performing_agents

CLAIM_AMOUNT
AGENT_ID                
AGENT00006         22400
AGENT00014          7400
AGENT00016         27000
AGENT00085         25700
AGENT00088         27700
AGENT00091         30000
AGENT00098         28200
AGENT00134         22000
AGENT00149         28900
AGENT00156         13200
AGENT00172         22200
AGENT00185         16100
AGENT00186         27700
AGENT00188         28400
AGENT00197         18500
AGENT00252         20000
AGENT00263         11500
AGENT00275         16200
AGENT00279         28000
AGENT00300         12600
AGENT00310         28800
AGENT00316         22200
AGENT00351         21700
AGENT00425         17800
AGENT00442         30300
AGENT00536         20000
AGENT00562         21100
AGENT00571         24200
AGENT00585         16600
AGENT00590         22200
AGENT00604          5500
AGENT00621          5300
AGENT00628          6100
AGENT00672         21300
AGENT00689         13700
AGENT00706         10800
AGENT00721          9000
AGENT00732         11300
AGENT00773         15000
AGENT00780         11300
AGENT00790          6000
AGENT00804         20000
AGENT00805         22600
AGENT00830         23500
AGENT00841         17800
AGENT00856         24200
AGENT00894         27600
AGENT00910         15500
AGENT00969         28300
AGENT00985         30000
AGENT01021         24300
AGENT01037          5600
AGENT01052         28000
AGENT01077          4200
AGENT01107         24700
AGENT01128         28000
AGENT01131         27500
AGENT01151         28000
AGENT01154           800
AGENT01179         27300

**Task #12**

Business wants to find all Suspicious Employees (Agents).
IF TOTAL CLAIM AMOUNT which meet below criteria is >= 15000 THEN AGENT is classified
as Suspicious ELSE Not
CLAIM_STATUS = Approved AND CUSTOMER_RISK_SEGMENTATION = High
AND INCIDENT_SEVERITY = “Major Loss”
If Suspicious, then 1 ELSE 0. Find mean of this column

In [60]:
criteria = (view['CLAIM_STATUS']=='A')&(view['RISK_SEGMENTATION']=='H')&(view['INCIDENT_SEVERITY']=='Major Loss')

In [65]:
z = view[criteria]
z

TXN_DATE_TIME TRANSACTION_ID CUSTOMER_ID POLICY_NUMBER  \
8     2020-06-01 00:00:00    TXN00000009   A00004671   PLC00006886   
75    2020-06-03 00:00:00    TXN00000076   A00009078   PLC00001505   
76    2020-06-03 00:00:00    TXN00000077   A00000804   PLC00006719   
92    2020-06-04 00:00:00    TXN00000093   A00007477   PLC00008821   
99    2020-06-04 00:00:00    TXN00000100   A00007283   PLC00001227   
...                   ...            ...         ...           ...   
9826  2021-06-24 00:00:00    TXN00009827   A00006637   PLC00000398   
9838  2021-06-24 00:00:00    TXN00009839   A00003275   PLC00009400   
9855  2021-06-25 00:00:00    TXN00009856   A00001546   PLC00000061   
9948  2021-06-29 00:00:00    TXN00009949   A00003202   PLC00000643   
9958  2021-06-29 00:00:00    TXN00009959   A00008989   PLC00006292   

     POLICY_EFF_DT     LOSS_DT   REPORT_DT INSURANCE_TYPE  PREMIUM_AMOUNT  \
8       2019-06-27  2020-05-21  2020-05-26          Motor           82.14   
75      2014-08-15  2020-05-17  2020-05-22           Life           67.88   
76      2011-07-09  2020-05-14  2020-05-19           Life           68.72   
92      2015-06-05  2020-05-23  2020-05-28         Health          116.27   
99      2014-09-25  2020-05-30  2020-05-31         Mobile           11.34   
...            ...         ...         ...            ...             ...   
9826    2017-06-26  2021-06-09  2021-06-14       Property          125.63   
9838    2013-10-31  2021-06-09  2021-06-14         Health          199.52   
9855    2017-01-27  2021-06-21  2021-06-22          Motor           87.78   
9948    2014-09-02  2021-06-24  2021-06-25       Property          113.24   
9958    2018-01-12  2021-06-09  2021-06-14         Health          112.88   

      CLAIM_AMOUNT  ... EMP_ROUTING_NUMBER     EMP_ACCT_NUMBER  \
8             5000  ...           92228117  VHVZ90875461509264   
75           59000  ...           33697833  FMCB37381150648042   
76           60000  ...           67659753  BMAQ41191552040977   
92            7000  ...           58706314  IGXL71581076733788   
99             200  ...           28574903  CEIG07819775081568   
...            ...  ...                ...                 ...   
9826         13000  ...           94644854  OEIB44255461327063   
9838          3000  ...           78720244  BAJN34272673771226   
9855          4000  ...           53733638  MTDS80945137889409   
9948         35000  ...          113432604  GZAY41352995514708   
9958          2000  ...          106717925  MWGP08083997541638   

                    VENDOR_NAME          ADDRESS_LINE1 ADDRESS_LINE2  \
8                    Hanson Inc  910 Arlington Terrace           NaN   
75    Gordon, Young and Webster         82 Queen Court           NaN   
76                Walker-Ingram        4770 Balboa Way           NaN   
92                  Sanchez LLC      154 Lyness Street           NaN   
99                          NaN                    NaN           NaN   
...                         ...                    ...           ...   
9826                        NaN                    NaN           NaN   
9838                 Dennis LLC      511 Norton Avenue           NaN   
9855                        NaN                    NaN           NaN   
9948   Reed, Bartlett and Miles   37 East Broad Street           NaN   
9958                Mcclure Ltd    243 Kentucky Avenue           NaN   

              CITY STATE POSTAL_CODE  COLOCATION  ELIGIBLE_FOR_DISCOUNT  
8     Fayetteville    AR     72701.0       False                      0  
75      Manchester    CT      6040.0       False                      0  
76         Fremont    CA     94536.0       False                      0  
92      Manchester    CT      6040.0       False                      0  
99             NaN   NaN         NaN       False                      0  
...            ...   ...         ...         ...                    ...  
9826           NaN   NaN         NaN       False                      

In [66]:
suspicion_data = z.groupby('AGENT_ID').sum()

In [84]:
suspiciou_agents = (suspicion_data[suspicion_data['CLAIM_AMOUNT']>=15000].index)

In [85]:
suspiciou_agents

Index(['AGENT00001', 'AGENT00003', 'AGENT00011', 'AGENT00021', 'AGENT00043',
       'AGENT00052', 'AGENT00076', 'AGENT00080', 'AGENT00097', 'AGENT00112',
       ...
       'AGENT01130', 'AGENT01142', 'AGENT01168', 'AGENT01175', 'AGENT01177',
       'AGENT01181', 'AGENT01182', 'AGENT01184', 'AGENT01188', 'AGENT01197'],
      dtype='object', name='AGENT_ID', length=149)

In [86]:
employee_data["SUSPICIOUS"]=0

In [88]:
employee_data.head()

AGENT_ID      AGENT_NAME DATE_OF_JOINING          ADDRESS_LINE1  \
0  AGENT00001       Ray Johns      1993-06-05       1402 Maggies Way   
1  AGENT00002   Angelo Borjon      2005-12-27         414 Tanya Pass   
2  AGENT00003  Candy Spellman      2003-09-02    606 National Street   
3  AGENT00004      Mary Smith      2004-09-23  235 Hugh Thomas Drive   
4  AGENT00005    Mildred Diaz      2011-06-21  3426 Broadview Street   

  ADDRESS_LINE2              CITY STATE  POSTAL_CODE  EMP_ROUTING_NUMBER  \
0           NaN  Waterbury Center    VT         5677            34584958   
1           NaN       Panama City    FL        32404           107363763   
2          #306      Fayetteville    AR        72701            81744097   
3           NaN       Panama City    FL        32404            67563771   
4           NaN        Montgomery    AL        36110           114951317   

      EMP_ACCT_NUMBER  SUSPICIOUS_agents  SUSPICIOUS  
0  HKUN51252328472585                  0           0  
1  OPIS19290040088204                  0           0  
2  YSCJ67489688482590                  0           0  
3  ZANG21285355574581                  0           0  
4  DZFS82244494451134                  0           0

In [90]:
suspicion_emp = employee_data['AGENT_ID'].isin(suspiciou_agents)

In [91]:
suspicion_emp 

0        True
1       False
2        True
3       False
4       False
        ...  
1195    False
1196     True
1197    False
1198    False
1199    False
Name: AGENT_ID, Length: 1200, dtype: bool

In [93]:
employee_data.loc[suspicion_emp,'SUSPICIOUS']=1

In [94]:
employee_data['SUSPICIOUS'].mean()

0.12416666666666666

The mean of suspicious column = 0.12416